src/rgw/rgw_aws_lambda.cc

```diff
@@ -0,0 +1,209 @@
+      *body << msg->payload;
+      invokeRequest.SetBody(body);
+      invokeRequest.SetContentType("application/json");
+      auto response = client->Invoke(invokeRequest);
+      if (!response.IsSuccess()) {
+        std::cout << response.GetError().GetMessage() << std::endl;
+      } else {
+        Aws::Lambda::Model::InvokeResult &result = response.GetResult();
+        Aws::IOStream &pl = result.GetPayload();
+        Aws::String returnedData;
+        std::getline(pl, returnedData);
+        std::cout << "Success " << returnedData << "\n" << std::endl;
+      }
+    }
+
+//    Works on messages in the queque
+    void run() {
+      while (!stopped) {
+        const auto count = messageQueue.consume_all(
+                std::bind(&Manager::publish_internal, this, std::placeholders::_1));
+        if (count == 0) {
+          std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+        }
+      }
+    }
+
+  public:
+    Manager(size_t _max_clients, size_t max_queue_size, CephContext *_cct) : messageQueue(max_queue_size),
+                                                                             runner(&Manager::run, this),
+                                                                             stopped(false), cct(_cct),
+                                                                             clientList(_max_clients) {}
+
+    Aws::Lambda::LambdaClient *connect(const std::string &accessKey,
+                                       const std::string &accessSecret,
+                                       const std::string &arn,
+                                       const std::string &caPath,
+                                       bool useSSL) {
+      if (stopped) {
+        return nullptr;
+      }
+      const client_id_t id(accessKey, accessSecret, arn, useSSL);
+      std::lock_guard lock(clientsLock);
+      const auto it = clientList.find(id);
+      if (it != clientList.end()) {
+        return it->second;
+      }
+
+      // connection not found, creating a new one
+      Aws::Client::ClientConfiguration configuration;
+      if (useSSL) {
+        configuration.scheme = Aws::Http::Scheme::HTTPS;
+        configuration.verifySSL = true;
+      } else {
+        configuration.scheme = Aws::Http::Scheme::HTTP;
+        configuration.verifySSL = false;
+      }
+      Aws::Utils::ARN arn1((Aws::String(arn)));
+      configuration.region = Aws::String(arn1.GetRegion());
+      configuration.maxConnections = 25;
+      configuration.requestTimeoutMs = 3000;
+      configuration.connectTimeoutMs = 3000;
+      configuration.caPath = Aws::String(caPath);
+      Aws::Auth::AWSCredentials credentials;
+      credentials.SetAWSAccessKeyId(Aws::String(accessKey));
+      credentials.SetAWSSecretKey(Aws::String(accessSecret));
+      auto client = new Aws::Lambda::LambdaClient(credentials, configuration);
+      ceph_assert(client);
```

Comment:
is it possible that client would be "nullptr"? e.g. conf issue, network issue?
if so, we should not assert, instead, we should return nullptr (and log)